scripts/test.sh

```diff
@@ -71,13 +71,14 @@ test -z "$(goimports -l -d $TEST_DIRS | tee /dev/stderr)"
 # go test -race ./... - Lets disable for now
 
 # Run test coverage on each subdirectories and merge the coverage profile.
-echo "mode: count" > profile.cov
-
+echo "mode: count (legacy)" > profile.cov
```

Comment:
This behavior of testing legacy or small tests needs to split in the test file and be triggered through changes to the .travis.yml file so that either legacy tests are run or small tests are run. A single test run in travis should not run both. This will make it cleaner to see a distinction between which tests are run and has failures in travis (legacy vs small), and allow us to see an increase in small tests as more tests get committed to the repo.